### PR TITLE
Add options for disabling less useful tests

### PR DESCRIPTION
**Pull request by t4c1**
Add options for disabling:
 - buffer tests
 - out of place tests, where we have equivalent in-place tests

This is intended for faster local development. Disabling both reduces the number float tests from ~1300 to ~400.

## Checklist

Tick if relevant:

* [N/A ] New files have a copyright
* [N/A ] New headers have an include guards
* [x] API is documented with Doxygen
* [N/A ] New functionalities are tested
* [x] Tests pass locally
* [x] Files are clang-formatted

### CODE REVIEW

**Review by Rbiessy — DISMISSED** [review]

**Comment by AD2605** [comment]
> I think we don't even rebuild kernels for USM and buffers after all. From the times I have seen in ctest the buffers tests never seem to JIT compile kernels. This may be thanks to the way we always get a pointer from the accessor so the `parallel_for` captures the same types.
> I wonder how much time we gain by disabling the 2 options when we use `SYCL_CACHE_PERSISTENT=1` though.

The kernel does not get jit compiled because USM tests are run first, and that's when all the kernels will be built, USM/Buffer/Layouts etc.. and then the runtime will cache it.
With this PR, even though the tests are disabled, we would still be building the kernels for type BUFFER ([see here](https://github.com/codeplaysoftware/portFFT/blob/c71814f5dc76ed3a6eadfef9d99f6a9729fa18fd/src/portfft/utils.hpp#L64)). If we disable building the Buffer kernels, we would be building 8 less kernels per test size. 

But its possible I have misunderstood the scope of this PR.

**Comment by Rbiessy** [comment]
Oh yes you're right, I forgot we were buffer and USM kernels there. I agree that we should disable building the buffer kernel in that case!

**Comment by t4c1** [comment]
Disabled JITing of buffers kernels when the option is disabled.

File: src/portfft/descriptor.hpp [hunk 1]
```diff
@@ -1094,24 +1094,28 @@ class committed_descriptor {
   template <direction Dir, typename TIn, typename TOut>
   sycl::event dispatch_direction(const TIn& in, TOut& out, const TIn& in_imag, TOut& out_imag,
                                  complex_storage used_storage, const std::vector<sycl::event>& dependencies = {}) {
-    if (used_storage != params.complex_storage) {
-      if (used_storage == complex_storage::SPLIT_COMPLEX) {
+    if constexpr (!PORTFFT_ENABLE_BUFFER_BUILDS && (!std::is_pointer_v<TIn> || !std::is_pointer_v<TOut>)) {
+      throw invalid_configuration("Buffer interface can not be called when buffer builds are disabled.");
+    } else {
```
Comment (Rbiessy) [inline src/portfft/descriptor.hpp:1099]: The `else` is not needed, I find that it is increasing indentation and making a larger diff for no good reason. How about removing it?

Comment (t4c1) [reply src/portfft/descriptor.hpp]: done

**Review by AD2605 — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -83,6 +85,14 @@ endif()
 if(${PORTFFT_USE_SCLA})
   target_compile_definitions(portfft INTERFACE PORTFFT_USE_SCLA)
 endif()
+if(${PORTFFT_ENABLE_BUFFER_BUILDS})
+  target_compile_definitions(portfft INTERFACE PORTFFT_ENABLE_BUFFER_BUILDS=1)
+else()
+  target_compile_definitions(portfft INTERFACE PORTFFT_ENABLE_BUFFER_BUILDS=0)
+endif()
```
Comment (Rbiessy) [inline CMakeLists.txt:92]: Sorry, just another thought: I don't think there is any reason to set the values `0` or `1` to `PORTFFT_ENABLE_BUFFER_BUILDS`, we can just define it or not define it like most of our other macros.

Comment (t4c1) [reply CMakeLists.txt]: done

**Review by Rbiessy — DISMISSED** [review]

**Review by AD2605 — DISMISSED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -39,6 +39,8 @@ option(PORTFFT_BUILD_TESTS "Whether to enable building tests" OFF)
 option(PORTFFT_BUILD_BENCHMARKS "Whether to enable building benchmarks" OFF)
 option(PORTFFT_VERIFY_BENCHMARKS "Whether to verify FFT results during benchmarking. Verifies in the first iteration only" OFF)
 option(PORTFFT_ENABLE_DOUBLE_BUILDS "Enable building tests and benchmarks using double precision" ON)
+option(PORTFFT_ENABLE_BUFFER_BUILDS "Enable building tests with buffers" ON)
+option(PORTFFT_ENABLE_OOP_BUILDS "Enable building testswith out-of-place configuration where an equivalent in-place configuration is supported" ON)
```
Comment (hjabird) [inline CMakeLists.txt:43]:
```suggestion
option(PORTFFT_ENABLE_OOP_BUILDS "Enable building tests with out-of-place configuration where an equivalent in-place configuration is supported" ON)
```
I don't think its worth the delays that will be caused by fixing this though.

**Review by hjabird — APPROVED** [review]
LGTM.

**Review by Rbiessy — APPROVED** [review]

**Review by hjabird — APPROVED** [review]

**Pull request merged**